pubsub/aws/snssqs/subscription_mgmt.go

```diff
@@ -0,0 +1,192 @@
+package snssqs
+
+import (
+	"context"
+	"sync"
+
+	"github.com/puzpuzpuz/xsync/v2"
+
+	"github.com/dapr/components-contrib/pubsub"
+	"github.com/dapr/kit/logger"
+)
+
+type (
+	SubscriptionAction int
+)
+
+const (
+	Subscribe SubscriptionAction = iota
+	Unsubscribe
+)
+
+var (
+	subscriptionMgmtInst *SubscriptionManager
```

Comment:
No singletons please. Each component should be completely self-contained, as one could instantiate multiple components.

Reply:
cool. maybe I'm too defensive